hazelcast-client-new/src/main/java/com/hazelcast/client/proxy/ClientMapProxy.java

```diff
@@ -1233,15 +1234,41 @@ public boolean isEmpty() {
 
     @Override
     public void putAll(Map<? extends K, ? extends V> m) {
-        Map<Data, Data> map = new HashMap<Data, Data>();
+        ClientPartitionService partitionService = getContext().getPartitionService();
+        Map<Integer, Map<Data, Data>> entryMap = new HashMap<Integer, Map<Data, Data>>(partitionService.getPartitionCount());
+        
         for (Entry<? extends K, ? extends V> entry : m.entrySet()) {
+            checkNotNull(entry.getKey(), NULL_KEY_IS_NOT_ALLOWED);
```

Comment:
I dont think it make sense to iterate same thing twice just for that. It will slowdown the best case scenario.
And as a defence ConcurentHashMap.putAll in java does the same. Puts some and then throw null pointer exception if came across null.